.github/workflows/CI.yml

```diff
@@ -123,6 +123,25 @@ jobs:
           fi
         shell: bash
 
+  coverage:
+    runs-on: ubuntu-latest
+    env:
+      CARGO_TERM_COLOR: always
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        run: rustup update stable
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Generate code coverage
+        run: cargo llvm-cov --all-features --workspace --lcov --output-path lcov.info
```

Comment:
What's the point of the Makefile if not to be used here 😅 ?

Reply:
Yeah I can totally add a new Makefile command (not sure the terminology?) for this.

The make file one generates html which is more for humans, just did it so people didn't have to worry about installing the tools and finding the right args themselves.

Reply:
> a new Makefile command (not sure the terminology?)

"Rule" is the word you're looking for. A rule has a _target_ and a _recipe_, and possible _pre-requisites_. 

> The make file one generates html which is more for humans

No worries. That works fine. We could merge the two, to help reproducing what the CI does locally more easily.